### PR TITLE
Refactor test files to allow extensions

### PR DESCRIPTION
**Pull request by farshidtz**

### CODE REVIEW

**Comment by farshidtz** [comment]
@MonicaisHer, please review

**Comment by farshidtz** [comment]
@MonicaisHer, please review. It would be easier if you check commit by commit.

File: tests/device-mqtt/network_interface_test.go [hunk 1]
```diff
@@ -12,20 +12,18 @@ var port = []string{"59982"}
...
 func setupSubtestNetworkInterface(t *testing.T) {
 	t.Log("[SUBTEST SETUP]")
-	stdout, stderr, err := utils.Command("sudo snap start --enable edgex-device-mqtt.device-mqtt")
-	utils.CommandLog(t, stdout, stderr, err)
+	utils.Command(t, "sudo snap start --enable edgex-device-mqtt.device-mqtt")
 
-	err = utils.WaitServiceOnline(t, port)
-	utils.CommandLog(t, "", "", err)
+	err := utils.WaitServiceOnline(t, port)
+	require.NoError(t, err)
```
Comment (MonicaisHer) [inline tests/device-mqtt/network_interface_test.go:18]: How about:
```
require.NoError(t, err,"Error waiting service(s) online.")
```

File: utils/utils.go [hunk 2]
```diff
@@ -187,7 +192,7 @@ func PortConnectionLocalhost(t *testing.T, ports []string) bool {
 				isOpen = true
 			} else {
 				isOpen = false
-				CommandLog(t, "", "", err)
+				require.NoError(t, err)
```
Comment (MonicaisHer) [inline utils/utils.go:195]: How about:
```
require.NoError(t, err,"Error listening port(s).")

File: utils/utils.go [hunk 3]
```diff
@@ -77,13 +81,14 @@ func CommandLog(t *testing.T, stdout string, stderr string, err error) {
 		}
 		if stderr != "" {
 			if err != nil {
+				// fatal error, non-zero exit
 				t.Error(stderr)
```
Comment (MonicaisHer) [inline utils/utils.go:85]: How about:
```
t.Fatal(stderr)
```

Comment (farshidtz) [reply utils/utils.go:85]: Yes, that was a mistake.

Comment (farshidtz) [reply utils/utils.go]: Actually it was correct. It should only log the error. Fatal will exit it right away which isn't what we want here.

**Pull request merged**